01-ruby-fundamentals/exercises/oo-library.md

```diff
@@ -0,0 +1,78 @@
+# OOD Exercise: Library Book
+
+Fact: Libraries have a lot of books.
+
+If we want to make a program that tracks all the books in a library, we can make the process much more efficient by making a class for a book, much like a template. That way we can more easily make and keep track of all the different books.
+
+
+## Baseline
+#### A book should have:
+  - title
+  - author
+  - description
+  - call number
+  - year published
+  - number of copies available to check-out
+  - list of names of people who have checked-out this book
+  - list of names of people who currently have book checked-out
+
+#### The book should be able to:
+- be checked-out
+- be checked-in
+- add more copies
+
+## Get to Work!
+1. Create a new ruby file
+1. Make a class for a library book
+1. Have your Book set the appropriate attributes on initialize
+1. Create the appropriate reader & writer methods for your class' attributes
+1. Have you tested yet? By now you should have made tests that:
+  - Can create new instances of your book class
+  - Pass in the appropriate attributes when creating the class
+  - Can call the appropriate attributes outside of the class
+1. Create a custom method for each action a book should be able to do. Make sure to test each method.
```

Comment:
The numbering in the rendered version seems to be messed up because of the bullets above